internal/controllers/clusterextension_controller.go

```diff
@@ -273,13 +273,12 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 
 	switch unpackResult.State {
 	case rukpaksource.StatePending:
-		setStatusUnpackPending(ext, unpackResult.Message)
-		setInstalledStatusConditionUnknown(ext, "installation has not been attempted as unpack is pending")
+		setStatusInstallFalseUnpackFailed(ext, unpackResult.Message)
```

Comment:
Unpack state is pending, but we're setting `Unpacked=False, Failed`?

Reply:
Well this was changed to acknowledge that we've switched to direct image registry and that as far as opr-ctrl is concerned we can't really be in an Unknown state, we're either failed or not. Perhaps we should just not process StatePending here?

Reply:
I'm not sure we should write this code with specific knowledge of the underlying `Source` implementation. In theory, the source could be switched out to a different implementation that _can_ return `StatePending`. I've been tinkering with an async direct image client, so this is not entirely hypothetical.

More broadly, I wonder if we even need an `Unpacked` condition type? I don't see it in the diagram in the RFC. The only reason we are unpacking a bundle is so that we can install it. So can we rollup unpacking status into Progressing and/or Installed conditions?

Reply:
That's farther than the RFC went, but I think it's in the same spirit. I think we can roll it into `TypeInstalled`. I think TypeProgressing would actually be adding a new type?

Reply:
as noted above there is no `TypeProgressing` currently, if we want to eliminate `TypeUnpacked` in favor of `TypeInstalled` 
@joelanford, these two helper methods encapsulate all the places changes would be needed:
```go
func setStatusUnpackFailed(ext *ocv1alpha1.ClusterExtension, message string) {
	ext.Status.InstalledBundle = nil
	apimeta.SetStatusCondition(&ext.Status.Conditions, metav1.Condition{
		Type:               ocv1alpha1.TypeUnpacked,
		Status:             metav1.ConditionFalse,
		Reason:             ocv1alpha1.ReasonUnpackFailed,
		Message:            message,
		ObservedGeneration: ext.GetGeneration(),
	})
}

func setStatusUnpacked(ext *ocv1alpha1.ClusterExtension, message string) {
	apimeta.SetStatusCondition(&ext.Status.Conditions, metav1.Condition{
		Type:               ocv1alpha1.TypeUnpacked,
		Status:             metav1.ConditionTrue,
		Reason:             ocv1alpha1.ReasonUnpackSuccess,
		Message:            message,
		ObservedGeneration: ext.GetGeneration(),
	})
}
```

So on the fail case we'd want `TypeInstalled` as `ConditionUnknown` and same on the success case. The reason would be the difference. But I wonder if the success case is a bit odd: `TypeInstalled` is Uknown and the reason is Unpack success?

Reply:
I think `TypeInstalled` should literally just be based on: "is there a helm release secret with status `deployed`? if so, Installed=True. If not "Installed=False". If there was an error looking up helm release secrets (other than "not found"), then "Installed=Unknown".

Nothing else should play into it.

Reply:
I created https://github.com/operator-framework/operator-controller/issues/1027 to capture this need. I'd like to keep this PR limited to the RFC covered items.